conda/core/initialize.py

```diff
@@ -621,9 +627,9 @@ def run_plan_elevated(plan):
             from ..common.os.windows import run_as_admin
             temp_path = None
             try:
-                with NamedTemporaryFile('w+b', suffix='.json', delete=False) as tf:
+                with Utf8NamedTemporaryFile('w+', suffix='.json', delete=False) as tf:
                     # the default mode is 'w+b', and universal new lines don't work in that mode
-                    tf.write(ensure_binary(json.dumps(plan, ensure_ascii=False)))
+                    tf.write(plan)
```

Comment:
Here, `type(plan)` is `list`, so if by some chance this code path is hit, there is a traceback.
PY3:
```
(base) C:\Users\Wani\repos\aggregate>conda init

# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<

    Traceback (most recent call last):
      File "C:\Users\Wani\repos\conda\conda\exceptions.py", line 1003, in __call__
        return func(*args, **kwargs)
      File "C:\Users\Wani\repos\conda\conda\cli\main.py", line 84, in _main
        exit_code = do_call(args, p)
      File "C:\Users\Wani\repos\conda\conda\cli\conda_argparse.py", line 82, in do_call
        exit_code = getattr(module, func_name)(args, parser)
      File "C:\Users\Wani\repos\conda\conda\cli\main_init.py", line 52, in execute
        anaconda_prompt, args.reverse)
      File "C:\Users\Wani\repos\conda\conda\core\initialize.py", line 111, in initialize
        run_plan_elevated(plan2)
      File "C:\Users\Wani\repos\conda\conda\core\initialize.py", line 632, in run_plan_elevated
        tf.write(plan)
      File "C:\Users\Wani\scratch\dev3\lib\tempfile.py", line 481, in func_wrapper
        return func(*args, **kwargs)
    TypeError: write() argument must be str, not list
```

PY2:
```
(C:\Users\Wani\scratch\dev) C:\Users\Wani\repos\conda>conda init
No handlers could be found for logger "conda._vendor.auxlib.packaging"
WARNING: The conda.compat module is deprecated and will be removed in a future release.

# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<

    Traceback (most recent call last):
      File "c:\users\wani\repos\conda\conda\exceptions.py", line 1003, in __call__
        return func(*args, **kwargs)
      File "c:\users\wani\repos\conda\conda\cli\main.py", line 84, in _main
        exit_code = do_call(args, p)
      File "c:\users\wani\repos\conda\conda\cli\conda_argparse.py", line 82, in do_call
        exit_code = getattr(module, func_name)(args, parser)
      File "c:\users\wani\repos\conda\conda\cli\main_init.py", line 52, in execute
        anaconda_prompt, args.reverse)
      File "c:\users\wani\repos\conda\conda\core\initialize.py", line 105, in initialize
        run_plan_elevated(plan1)
      File "c:\users\wani\repos\conda\conda\core\initialize.py", line 628, in run_plan_elevated
        tf.write(plan)
      File "C:\Users\Wani\scratch\dev\lib\codecs.py", line 369, in write
        data, consumed = self.encode(object, self.errors)
    TypeError: coercing to Unicode: need string or buffer, list found
```

However, if we make it dump json, then it fails to serialize one of the arguments to one of the functions:
```
{  
   'function':'init_cmd_exe_registry',
   'kwargs':{  
      'target_path':'HKEY_CURRENT_USER\\Software\\Microsoft\\Command Processor\\AutoRun',
      'conda_prefix':'C:\\Users\\Wani\\scratch\\dev3',
      'reverse':<auxlib._Null object at 0x0000019F60A0E080> <---------- 😭 
   },
   'result':'no change'
}
```

Reply:
I think the right answer is to fix the _Null to a JSON-serializable type, or perhaps teach the JSON serializer how to serialize that particular type, and to dump to JSON.

Reply:
Okay, this diff seems to make it work:
```patch
diff --git a/conda/core/initialize.py b/conda/core/initialize.py
index 1fc6caa53..451e9651a 100644
--- a/conda/core/initialize.py
+++ b/conda/core/initialize.py
@@ -629,7 +629,8 @@ def run_plan_elevated(plan):
             try:
                 with Utf8NamedTemporaryFile('w+', suffix='.json', delete=False) as tf:
                     # the default mode is 'w+b', and universal new lines don't work in that mode
-                    tf.write(plan)
+                    tf.write(json.dumps(plan, ensure_ascii=False,
+                                        default=lambda x: x.__dict__))
                     temp_path = tf.name
                 python_exe = '"%s"' % abspath(sys.executable)
                 hinstance, error_code = run_as_admin((python_exe, '-m',  'conda.core.initialize',
```